lib/publiccloud/utils.pm

```diff
@@ -885,4 +887,129 @@ sub upload_asset_on_remote {
     $instance->ssh_assert_script_run($mv_cmd);
 }
 
+
+=head2 $instance->zypper_call
+
+    $instance->zypper_call($command [, exitcode => $exitcode] [, timeout => $timeout];
+
+Function wrapping zypper command for remote execution via ssh; not for tunnelling.
+Implements similar routine as lib/utils::zypper_call, but for remote execution on publiccloud instances.
+
+Usage example:
+    $instance->zypper_call("up", exitcode => [0,102,103], timeout => 300);
+
+=cut
+
+sub zypper_call {
+    my $instance = shift;
+    my %args = testapi::compat_args({cmd => undef}, ['cmd'], @_);
+    # check command line:
+    die("Parameters missing") unless ($args{cmd});
+    die 'Exit code is from PIPESTATUS[0], not grep' if $args{cmd} =~ /^((?!`).)*\| ?grep/;
+    #
+    $args{rc_only} = 1;
```

Comment:
Is this used?

Reply:
I think Felix is right we should not allow caller of `zypper_call_remote` to change it to 0 because than it will break code flow which you created

Reply:
yeah Maurizio was right !